app/javascript/controllers/visibility_controller.js

```diff
@@ -0,0 +1,41 @@
+/* eslint no-await-in-loop:0, no-restricted-syntax:0 */
+import { Controller } from '@hotwired/stimulus';
+import { useClickOutside } from 'stimulus-use';
+import { enter, leave } from 'el-transition';
+
+// transitions are optional, but to use them add the el-transition data attributes listed here to content targets:
+// https://github.com/mmccall10/el-transition#dataset-attributes
+export default class VisibilityController extends Controller {
```

Comment:
Open to suggestions on the naming for this. I'm not 100% settled on `visibility`. 

I mostly went with this for the clear API:
* `visibility#on`
* `visibility#off`
* `visibility#toggle`

Reply:
I can't think of a better name mate.